Lib/test/test_strptime.py

```diff
@@ -466,6 +466,11 @@ def test_mar1_comes_after_feb29_even_when_omitting_the_year(self):
                 time.strptime("Feb 29", "%b %d"),
                 time.strptime("Mar 1", "%b %d"))
 
+    def test_timezone_offset_cannot_parse_z(self):
+        # Check ValueError is raised when matching "z" (ordinary letter)
+        # with "%z" (timezone offset)
+        self.assertRaises(ValueError, _strptime._strptime_time, "z", "%z")
```

Comment:
Two things here:

1. I much prefer the context manager version of `assertRaises`, since it's much clearer what calls are being tested.
2. It would be preferable to test this using only the public interface. I realize that there are other uses of `_strptime._strptime_time` in this file, but going forward I'd like to keep the tests focused on the public interface, so can you try and exercise this using `time.strptime` and/or `datetime.strptime`?